src/Offcanvas.php

```diff
@@ -4,235 +4,554 @@
     {
         $new = clone $this;
-        $new->headerOptions = $options;
+        $new->cssClasses = $class;
 
         return $new;
     }
 
     /**
-     * Set/remove offcanvas title
+     * Sets the HTML attributes for the header.
+     *
+     * @param array $attributes Attribute values indexed by attribute names.
+     *
+     * @return self A new instance with the specified header attributes.
+     *
+     * @see {\Yiisoft\Html\Html::renderTagAttributes()} for details on how attributes are being rendered.
      */
-    public function title(?string $title): self
+    public function headerAttributes(array $attributes): self
     {
         $new = clone $this;
-        $new->title = $title;
+        $new->headerAttributes = $attributes;
 
         return $new;
     }
 
     /**
-     * The HTML attributes for the widget title tag. The following special options are recognized.
+     * Sets the ID.
+     *
+     * @param bool|string $id The ID of the component. If `true`, an ID will be generated automatically.
+     *
+     * @throws InvalidArgumentException if the ID is an empty string or `false`.
      *
-     * {@see Html::renderTagAttributes()} for details on how attributes are being rendered.
+     * @return self A new instance with the specified ID.
      */
-    public function titleOptions(array $options): self
+    public function id(bool|string $id): self
     {
         $new = clone $this;
-        $new->titleOptions = $options;
+        $new->id = $id;
 
         return $new;
     }
 
     /**
-     * The HTML attributes for the widget body tag. The following special options are recognized.
+     * Sets the placement.
      *
-     * {@see Html::renderTagAttributes()} for details on how attributes are being rendered.
+     * @param OffcanvasPlacement|string $placement The placement.
+     *
+     * @return self A new instance with the specified placement setting.
      */
-    public function bodyOptions(array $options): self
+    public function placement(OffcanvasPlacement|string $placement): self
     {
         $new = clone $this;
-        $new->bodyOptions = $options;
+        $new->placement = $placement;
 
         return $new;
     }
 
-    public function begin(): string
+    /**
+     * Sets the responsive size.
+     *
+     * @param Responsive|string $size The responsive size.
+     *
+     * @return self A new instance with the specified responsive size setting.
+     */
+    public function responsive(Responsive|string $size): self
     {
-        parent::begin();
+        $size = $size instanceof Responsive ? $size->value : $size;
 
-        $options = $this->options;
-        $bodyOptions = $this->bodyOptions;
-        $tag = ArrayHelper::remove($options, 'tag', 'div');
-        $bodyTag = ArrayHelper::remove($bodyOptions, 'tag', 'div');
+        $new = clone $this;
+        $new->responsive = match ($size) {
+            'sm', 'md', 'lg', 'xl', 'xxl' => $size,
```

Comment:
I'd get values from `Responsive`. This way you won't have any values hardcoded here.

Reply:
done.